GCRCatalogs/cosmodc2.py

```diff
@@ -0,0 +1,406 @@
+        for version_label in ('Major', 'Minor', 'MinorMinor'):
+            try:                                              
+                catalog_version.append(fh['/metaData/version' + version_label].value)
+            except KeyError:                                                         
+                break                                                                
+        catalog_version = StrictVersion('.'.join(map(str, catalog_version or (0, 0))))
+
+        #check cosmology
+        metakeys = fh['metaData'].keys()
+        if 'H_0' in metakeys and 'Omega_matter' in metakeys and 'Omega_b' in metakeys:
+            H0 = fh['metaData/H_0'].value
+            Om0 = fh['metaData/Omega_matter'].value
+            Ob0 = fh['metaData/Omega_b'].value
+            if  abs(H0 - self.cosmology.H0.value) > tol or abs(Om0 - self.cosmology.Om0) > tol or abs(Ob0 - self.cosmology.Ob0) > tol:
+                raise ValueError('Mismatch in cosmological parameters (H0:{}, Om0:{}, Ob0:{}) for healpix file {}'.format(H0, Om0, Ob0, healpix_file))
+
+        # check versions
+        config_version = StrictVersion(self.version)
+        if config_version != catalog_version:
+            raise ValueError('Catalog file version {} does not match config version {}'.format(catalog_version, config_version))
+        if StrictVersion(__version__) < config_version:
+            raise ValueError('Reader version {} is less than config version {}'.format(__version__, catalog_version))
+
+
+    def _get_skyarea_info(self):
+        self._skyarea = {}      
+        pattern = re.compile('\d+')
+        for healpix_file in self.healpix_pixel_files:
+            fh = h5py.File(healpix_file, 'r')        
+            healpix_name = os.path.splitext(os.path.basename(healpix_file))[0]
+            zlo, zhi, hpx = pattern.findall(healpix_name)                     
+            if hpx not in self._skyarea:                                      
+                self._skyarea[hpx] = {}                                       
+            if 'skyArea' in fh['metaData'].keys():                            
+                self._skyarea[hpx][zlo+'_'+zhi] = float(fh['metaData/skyArea'].value)
+            else:                                                                    
+                self._skyarea[hpx][zlo+'_'+zhi] = np.rad2deg(np.rad2deg(4.0*np.pi/768.))
+            fh.close()
+
+        sky_area = 0.0
+        for hpx in self._skyarea:
+            sky_area = sky_area + max([self._skyarea[hpx][z] for z in self._skyarea[hpx]])
+
+        return sky_area
+
+
+    def _generate_native_quantity_list(self):
+        return self._native_quantities
+
+    def _iter_native_dataset(self, native_filters=None):
+        for healpix in self.healpix_pixels:
+
+            if native_filters is not None:
+                fargs = dict(healpix_pixel=healpix)
+                if not all(f[0](*(fargs[k] for k in f[1:])) for f in native_filters if set(f[1:]) == set(fargs)):
```

Comment:
I know this PR is not yet ready to be reviewed, but just a heads-up, this line would no longer work with the newer version of GCR that we are using. Change it to:
```python
if not native_filters.check_scalar(fargs):
```

Reply:
@yymao Sorry, but do you mean replace line 369 with your substitution?  But fargs isn't defined yet so I would have to move line 370 up, right?

Reply:
I meant replacing 371. Or you can combine 369-371 to one line:
```python
if native_filters is not None and not native_filters.check_scalar(dict(healpix_pixel=healpix)):
```